Yoast/wordpress-seo
# Consolidate scripts and script data into single sources per screen.

## Context

⚠️ Don't merge. Incompatibilities found. ⚠️ 
Right now, we enqueue many scripts and expose dozens of so called `l10n` objects to our JavaScript. This PR is a first stab at consolidating the JS into fewer sources and consolidating all the different localization objects into a single variable `wpseoScriptData`. This should make it easier to see which data is localized on the post edit, term edit pages and settings pages.

## Summary


This PR can be summarized in the following changelog entry:

* Consolidates 10 scripts into 3.
* Gets rid of the following global variables in JavaScript and consolidates them into `wpseoScriptData`:
  * `wpseoReplaceVarsL10n` was moved to `wpseoScriptData.analysis.plugins.replaceVars`
  * `wpseoSearchAppearance` was moved to `wpseoScriptData.searchAppearance`
  * `wpseoSelect2Locale` was moved to `wpseoScriptData.userLanguageCode`
  * `wpseoMediaL10n` was moved to `wpseoScriptData.media`
  * `wpseoPostScraperL10n` was moved to `wpseoScriptData.metabox` but kept variable available for backwards compatibility.
  * `wpseoAnalysisWorkerL10n` was moved to `wpseoScriptData.analysis.worker`
  * `wpseoShortcodePluginL10n` was moved to `wpseoScriptData.analysis.plugins.shortcodes`
  * `wpseoFeaturedImageL10n` was moved to `wpseoScriptData.featuredImage`
  * `wpseoTermScraperL10n` was moved to `wpseoScriptData.metabox` but kept variable available for backwards compatibility.

## Relevant technical choices:

* Introduces `post-edit.js`, `term-edit.js` and `settings.js`.
* Moves all YoastSEO.js plugins to `js/src/analysis/plugins`. These are now imported where they are used.
* Moves `search-appearance.js`, `wpseo-admin.js`, `wp-seo-admin-media.js`, `wp-seo-metabox.js`, `wp-seo-post-scraper.js`, `wp-seo-term-scraper.js`, `wpseo-metabox-image.js` and `wp-seo-primary-category.js` to `js/src/initializers`. These are now loaded in the new main JavaScript files.
* Most script data in `WPSEO_Metabox` and `WPSEO_Taxonomy` and `WPSEO_Config` is now localized on one file and consolidated into one variable `wpseoScriptData`. This makes it much easier to add script data down the line, without cluttering the global namespace with a myriad of poorly named l10n objects.

## Test instructions

This PR can be tested by following these steps:

* Check for no JS errors on post edit, term edit and settings pages.
* Please test with Premium and all Yoast add-ons enabled.
* Also test with products, Local SEO locations.

## Documentation

## Quality assurance

* [x] I have checked if any of our other plugins depend on any of the removed scripts. 
  * 😱 `admin-script` is still depended on by Local SEO.
  * 😱 `admin-media` is still depended on by Local SEO
  * 😱 `post-scraper` is still depended on by Yoast ACF glue
  * 😱 `term-scraper` is still depended on by Yoast ACF glue
* [x] I have checked if the removed variables are used in any of our other plugins:
  * ✅ `wpseoReplaceVarsL10n` is not used anywhere.
  * ✅ `wpseoSearchAppearance` is not used anywhere.
  * 🆗 `wpseoSelect2Locale` is still used in Yoast SEO Premium on the redirects settings page but localized separately there.
  * 🆗 `wpseoMediaL10n` is still used in the Yoast AMP plugin but localized separately there.
  * 😢 `wpseoPostScraperL10n` is still used in Yoast SEO Premium. In `post-edit.js`, I bring it back to provide backwards compatibility.
  * ✅ `wpseoAnalysisWorkerL10n` is not used anywhere.
  * ✅ `wpseoShortcodePluginL10n` is not used anywhere.
  * ✅ `wpseoFeaturedImageL10n` is not used anywhere.
  * 😢 `wpseoTermScraperL10n` is still used in Yoast SEO Premium. In `term-edit.js`, I bring it back to provide backwards compatibility.
